### PR TITLE
ADR 007: Solo machine signature construction

### PR DESCRIPTION
**Pull request by colin-axner**
## Description

closes: #1847 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [x] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [x] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [x] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: docs/architecture/adr-007-solomachine-signbytes.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+## Changelog
+* 2022-08-02: Initial draft
+
+## Status
+
+Accepted, applied in v6
+
+## Context
+
+The solo machine implemention up until ibc-go v6 constructed sign bytes using a `DataType` which described what type of data was being signed.
```
Comment (damiannolan) [inline docs/architecture/adr-007-solomachine-signbytes.md:12]:
```suggestion
The `06-solomachine` implemention up until ibc-go v6 constructed sign bytes using a `DataType` which described what type of data was being signed. 
```

File: docs/architecture/adr-007-solomachine-signbytes.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+## Context
+
+The solo machine implemention up until ibc-go v6 constructed sign bytes using a `DataType` which described what type of data was being signed. 
+This design decision arose from a misunderstanding of the security implications.
+It was noted that the proto definitions do not [provide uniqueness](https://github.com/cosmos/cosmos-sdk/pull/7237#discussion_r484264573) which is a necessity for ensuring two signatures over different data types can never be the same.
+What was missed is that the uniqueness is not provided by the proto definition, but by the usage of the proto definition. 
+The path provided by core IBC will be unique and is already encoded into the signature data.
+Thus two different paths with the same data values will encode differently which provides signatuer uniqueness. 
+
+Furthermore,  the current construction does not support the proposed changes in the spec repo to support [Generic Verification functions](https://github.com/cosmos/ibc/issues/684).
```
Comment (damiannolan) [inline docs/architecture/adr-007-solomachine-signbytes.md:19]:
```suggestion
Furthermore, the current construction does not support the proposed changes in the spec repo to support [Generic Verification functions](https://github.com/cosmos/ibc/issues/684).
```

**Review by damiannolan — APPROVED** [review]
Nice work! Thank you for putting this together ❤️

File: docs/architecture/adr-007-solomachine-signbytes.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+Accepted, applied in v6
+
+## Context
+
+The `06-solomachine` implemention up until ibc-go v6 constructed sign bytes using a `DataType` which described what type of data was being signed. 
+This design decision arose from a misunderstanding of the security implications.
+It was noted that the proto definitions do not [provide uniqueness](https://github.com/cosmos/cosmos-sdk/pull/7237#discussion_r484264573) which is a necessity for ensuring two signatures over different data types can never be the same.
+What was missed is that the uniqueness is not provided by the proto definition, but by the usage of the proto definition. 
+The path provided by core IBC will be unique and is already encoded into the signature data.
+Thus two different paths with the same data values will encode differently which provides signatuer uniqueness.
```
Comment (AdityaSripal) [inline docs/architecture/adr-007-solomachine-signbytes.md:17]:
```suggestion
Thus two different paths with the same data values will encode differently which provides signature uniqueness. 
```

**Review by AdityaSripal — APPROVED** [review]
Nice work!

**Review by seantking — APPROVED** [review]

File: docs/architecture/adr-007-solomachine-signbytes.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+## Changelog
+* 2022-08-02: Initial draft
+
+## Status
+
+Accepted, applied in v6
+
+## Context
+
+The `06-solomachine` implemention up until ibc-go v6 constructed sign bytes using a `DataType` which described what type of data was being signed.
```
Comment (charleenfei) [inline docs/architecture/adr-007-solomachine-signbytes.md:12]: nit
```suggestion
The `06-solomachine` implemention up until `ibc-go` v6 constructed sign bytes using a `DataType` which described what type of data was being signed. 
```

File: docs/architecture/adr-007-solomachine-signbytes.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+## Changelog
+* 2022-08-02: Initial draft
+
+## Status
+
+Accepted, applied in v6
+
+## Context
+
+The `06-solomachine` implemention up until `ibc-go` v6 constructed sign bytes using a `DataType` which described what type of data was being signed.
```
Comment (crodriguezvega) [inline docs/architecture/adr-007-solomachine-signbytes.md:12]:
```suggestion
The `06-solomachine` implemention up until ibc-go v6 constructed sign bytes using a `DataType` which described what type of data was being signed. 
```

**Review by crodriguezvega — APPROVED** [review]

**Pull request merged**